libpod/runtime_ctr.go

```diff
@@ -915,6 +915,44 @@ func (r *Runtime) evictContainer(ctx context.Context, idOrName string, removeVol
 	return id, cleanupErr
 }
 
+// DependMap populates of map of all containers depending on the specified container
+func (r *Runtime) DependMap(ctx context.Context, ctr *Container, ctrMap map[string]*Container) error {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
```

Comment:
Do we really need to lock? I _think_ we can _read_ the data in any case.

Reply:
Rewritten, and yes we will need the lock.

Reply:
This is just the runtime lock, we need to take the container lock (`c.lock.Lock()`) instead

Reply:
Also, the lock needs to be in `removeDepend()` - otherwise we don't take the lock when we recurse onto other containers.

Reply:
Can probably avoid `removeDepend` entirely given this, runtime lock is superfluous

Reply:
Please check the code again.